File: vscode4teaching-server/src/main/java/com/vscode4teaching/vscode4teachingserver/model/ExerciseStatus.java
package com.vscode4teaching.vscode4teachingserver.model;

public enum ExerciseStatus {
    NOT_STARTED(0, "Not started"),
    IN_PROGRESS(2, "In progress"),
    FINISHED(1, "Finished");

    private final int code;
    private final String asString;

    ExerciseStatus(int code, String asString) {
        this.code = code;
        this.asString = asString;
    }

    public int getCode() {
        return code;
    }

    @Override
    public String toString() {
        return asString;
    }
}

File: vscode4teaching-extension/src/model/serverModel/exercise/ExerciseStatus.ts
export namespace ExerciseStatus {
    export enum StatusEnum {
        NOT_STARTED = "NOT_STARTED",
        IN_PROGRESS = "IN_PROGRESS",
        FINISHED = "FINISHED"
    }

    export function toString(status: StatusEnum): string {
        switch (status) {
            case "NOT_STARTED":
                return "Not started";
            case "FINISHED":
                return "Finished";
            case "IN_PROGRESS":
                return "In progress";
            default:
                return "";
        }
    }
}
